### DISCUSSION TITLE
alternative name for composer.json

### DISCUSSION THREAD

**Discussion by rino1** [composer/composer#11400] [General]
````
$ cat .config/composer/config.json
{
    "repositories": {
        "abc": {
            "type": "path",
            "url": "abc"
        }
    }
}

$ cat abc/alternative-composer.json
{
    "name": "xyz/abc"
}

$ composer show -a xyz/abc
Package "xyz/abc" not found
````

mmmmmmmmmmhhhh...

````
$ mv -i abc/alternative-composer.json abc/composer.json

$ composer show -a xyz/abc
name     : xyz/abc
descrip. :
keywords :
versions : dev-master, dev-main
type     : library
homepage :
source   : []
dist     : [path] abc ...
names    : xyz/abc
````

Ok, but so there isn't a way to give composer.json an alternative name?

For example, by changing the .config/composer/config.json file in this way

````
{
    "repositories": {
        "abc": {
            "composer_alternative_filename": "alternative-composer.json",
            "type": "path",
            "url": "abc"
        }
    }
}
````

**Comment by stof** [comment]
if you talk about the root composer.json file on your machine, you can have a different name by setting the path in the `COMPOSER` env variable.
However, for shared packages, the file name must be `composer.json` (repository implementations don't support configuring the file, as that would make them a lot more complex)

**Reply by rino1** [reply]
thanks for info
